hudi-common/src/main/java/org/apache/hudi/common/model/PartialUpdateAvroPayload.java

```diff
@@ -0,0 +1,196 @@
+    } else {
+      schemaInstance = schemaRepo.get(schemaStringIn);
+    }
+    if (oldValue.recordBytes.length == 0) {
+      // use natural order for delete record
+      return this;
+    }
+
+    try {
+      GenericRecord indexedOldValue = (GenericRecord) oldValue.getInsertValue(schemaInstance).get();
+      Option<IndexedRecord> optValue = combineAndGetUpdateValue(indexedOldValue, schemaInstance, this.orderingVal.toString());
+      // Rebuild ordering value if required
+      String newOrderingVal = rebuildOrderingValMap((GenericRecord) optValue.get(), this.orderingVal.toString());
+      if (optValue.isPresent()) {
+        return new PartialUpdateAvroPayload((GenericRecord) optValue.get(), newOrderingVal);
+      }
+    } catch (Exception ex) {
+      return this;
+    }
+    return this;
+  }
+
+  public Option<IndexedRecord> combineAndGetUpdateValue(
+      IndexedRecord currentValue, Schema schema, String newOrderingValWithMappings) throws IOException {
+    Option<IndexedRecord> recordOption = getInsertValue(schema);
+    if (!recordOption.isPresent()) {
+      return Option.empty();
+    }
+
+    // Perform a deserialization again to prevent resultRecord from sharing the same reference as recordOption
+    GenericRecord resultRecord = (GenericRecord) getInsertValue(schema).get();
+    List<Schema.Field> fields = schema.getFields();
+
+    // newOrderingValWithMappings = _ts1:name1,price1=999;_ts2:name2,price2=;
+    for (String newOrderingFieldMapping : newOrderingValWithMappings.split(";")) {
+      String orderingField = newOrderingFieldMapping.split(":")[0];
+      String newOrderingVal = getNewOrderingVal(orderingField, newOrderingFieldMapping);
+      String oldOrderingVal = HoodieAvroUtils.getNestedFieldValAsString(
+          (GenericRecord) currentValue, orderingField, true, false);
+      if (oldOrderingVal == null) {
+        oldOrderingVal = "";
+      }
+
+      // No update required
+      if (oldOrderingVal.isEmpty() && newOrderingVal.isEmpty()) {
+        continue;
+      }
+
+      // Pick the payload with greatest ordering value as insert record
+      boolean isBaseRecord = false;
+      try {
+        if (Long.parseLong(oldOrderingVal) > Long.parseLong(newOrderingVal)) {
+          isBaseRecord = true;
+        }
+      } catch (NumberFormatException e) {
+        if (oldOrderingVal.compareTo(newOrderingVal) > 0) {
+          isBaseRecord = true;
+        }
+      }
+
+      // Initialise the fields of the sub-tables
+      GenericRecord insertRecord;
+      if (isBaseRecord) {
+        insertRecord = (GenericRecord) currentValue;
+        // resultRecord is already assigned as recordOption
+      } else {
+        insertRecord = (GenericRecord) recordOption.get();
+        GenericRecord currentRecord = (GenericRecord) currentValue;
+        fields.stream()
+            .filter(f -> newOrderingFieldMapping.contains(f.name()))
+            .forEach(f -> resultRecord.put(f.name(), currentRecord.get(f.name())));
```

Comment:
There is a problem. For example, table with schema (key, col1, tsA, col2, tsB)
partial write client A with schema: key, col1, tsA
partial write client B with schema: key, col2, tsB
current value belong to A with old value belong to B, so the current value should be updated. But here only update the tsA and tsB without update the col1 which doesn't belong to precombine field.

Reply:
precombine field over here looks like this “ _ts1:name1,price1=999;_ts2:name2,price2=”   it should contain col1 and col2